lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorManager.java

```diff
@@ -232,13 +232,15 @@ public void close() {
                     .setKillEvaluator(EvaluatorRuntimeProtocol.KillEvaluatorProto.newBuilder().build())
                     .build();
             sendEvaluatorControlMessage(evaluatorControlProto);
+            this.stateManager.setClosing();
+          } else {
+            this.stateManager.setKilled();
           }
-        } finally {
+        } catch (Exception e) {
```

Comment:
In both case that exception is occurred or not, the state is set to be `KILLED` beforehand.
At now, we should change it to `CLOSING` instead of `KILLED` if there is no exception and evaluator is running.
Therefore, setting the state to `KILLED` always is not proper way.

Also, I caught general exception because many kind of specific exception could occur during sending runtime protocol message.